src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -598,6 +603,9 @@ void QgsIdentifyResultsDialog::addFeature( QgsVectorLayer *vlayer, const QgsFeat
                                 ? QStringLiteral( " [%1]" ).arg( layItem->childCount() )
                                 : QString();
   layItem->setText( 0, QStringLiteral( "%1 %2" ).arg( vlayer->name(), countSuffix ) );
+  
+  
+  updateTextDisplay();
```

Comment:
This shouldn't be called here -- this method will be called many times if multiple features are identified, and it's iterating through everything in the list for every feature added. That'll add up quickly.